src/java.base/share/classes/java/security/Security.java

```diff
@@ -142,9 +142,13 @@ private static void initialize() {
         }
 
         /*[IF CRIU_SUPPORT]*/
-        // Check if CRIU checkpoint mode is enabled, if it is then reconfigure the security providers.
-        if (InternalCRIUSupport.isCheckpointAllowed()) {
+        // If -XX:-CRIUSecProvider is specified, we don't need to configure the CRIUSec provider.
+        if (InternalCRIUSupport.isCheckpointAllowed() && InternalCRIUSupport.enableCRIUSecProvider()) {
             CRIUConfigurator.setCRIUSecMode(props);
+            // Check if `-XX:+EnableFullSecurityEnv` is enabled, if it is then reconfigure the security providers.
+            if (InternalCRIUSupport.isEnableFullSecurityEnvInCRIU()) {
+                CRIUConfigurator.setCRIURestoreMode();
+            }
```

Comment:
This should be the same as in later streams (also `isEnableFullSecurityEnvInCRIU` doesn't exist).